CopilotKit/packages/backend/src/lib/openai-assistant-adapter.ts

```diff
@@ -117,18 +117,32 @@ export class OpenAIAssistantAdapter implements CopilotKitServiceAdapter {
   async getResponse(forwardedProps: any): Promise<CopilotKitResponse> {
     // copy forwardedProps to avoid modifying the original object
     forwardedProps = { ...forwardedProps };
-
+  
     const forwardMessages = forwardedProps.messages || [];
-
+  
     // Remove tools if there are none to avoid OpenAI API errors
     // when sending an empty array of tools
     if (forwardedProps.tools && forwardedProps.tools.length === 0) {
       delete forwardedProps.tools;
     }
-
+  
     // get the thread from forwardedProps or create a new one
     const threadId: string =
       forwardedProps.threadId || (await this.openai.beta.threads.create()).id;
+  
+    // Use createAndStream for user messages
+    if (
+      forwardMessages.length > 0 &&
+      forwardMessages[forwardMessages.length - 1].role === "user"
+    ) {
+      const run = await this.openai.beta.threads.runs.createAndStream(threadId, {
```

Comment:
you want to make the `createAndStream` change inside the existing `submitUserMessage` function. 

it encapsulates additional context injections and business logic